kratos/tests/strategies/schemes/test_schemes.cpp

```diff
@@ -0,0 +1,288 @@
+            ModelPart.SetBufferSize(3);
+            
+            ModelPart.AddNodalSolutionStepVariable(DISPLACEMENT);
+            ModelPart.AddNodalSolutionStepVariable(VELOCITY);
+            ModelPart.AddNodalSolutionStepVariable(ACCELERATION);
+            
+            auto pnode = ModelPart.CreateNewNode(1, 0.0, 0.0, 0.0);
+            
+            pnode->AddDof(DISPLACEMENT_X);
+            pnode->AddDof(DISPLACEMENT_Y);
+            pnode->AddDof(DISPLACEMENT_Z);
+            
+            ModelPart.GetProcessInfo()[DELTA_TIME] = DeltaTime;
+            
+            DoF.reserve(3);
+            DoF.push_back(pnode->pGetDof(DISPLACEMENT_X));
+            DoF.push_back(pnode->pGetDof(DISPLACEMENT_Y));
+            DoF.push_back(pnode->pGetDof(DISPLACEMENT_Z));
+            
+            // Set initial solution
+            const array_1d<double, 3> zero_vector = ZeroVector(3);
+            pnode->FastGetSolutionStepValue(DISPLACEMENT) = zero_vector;
+            pnode->FastGetSolutionStepValue(DISPLACEMENT, 1) = zero_vector;
+            pnode->FastGetSolutionStepValue(DISPLACEMENT, 2) = zero_vector;
+            pnode->FastGetSolutionStepValue(VELOCITY) = zero_vector;
+            pnode->FastGetSolutionStepValue(VELOCITY, 1) = zero_vector;
+            pnode->FastGetSolutionStepValue(VELOCITY, 2) = zero_vector;
+            pnode->FastGetSolutionStepValue(ACCELERATION) = zero_vector;
+            pnode->FastGetSolutionStepValue(ACCELERATION, 1) = zero_vector;
+            pnode->FastGetSolutionStepValue(ACCELERATION, 2) = zero_vector;
+            
+            DofsArrayType Doftemp;
+            Doftemp.reserve(DoF.size());
+            for (auto it= DoF.begin(); it!= DoF.end(); it++)
+            {
+                Doftemp.push_back( it->get() );
+            }
+            Doftemp.Sort();
+            
+            CompressedMatrix A = ZeroMatrix(3, 3);
+            Vector Dx = ZeroVector(3);
+            Vector b = ZeroVector(3);
+            
+            pScheme->Initialize(ModelPart);
+            
+            return Doftemp;
+        }
+     
+        /** 
+         * Checks if the Newmark scheme performs correctly the integration
+         */
+        
+        KRATOS_TEST_CASE_IN_SUITE(DisplacementNewmarkSchemeTest, KratosCoreSchemesFastSuite) 
+        {
+            constexpr double tolerance = 1e-6;
+            
+            ModelPart model_part("Main");
+            
+            typedef ResidualBasedNewmarkDisplacementScheme< SparseSpaceType, LocalSpaceType >  ResidualBasedNewmarkDisplacementSchemeType;
+            typename SchemeType::Pointer pscheme = typename SchemeType::Pointer( new ResidualBasedNewmarkDisplacementSchemeType() );
+            
+            const double delta_time = 1.0e-4;
+
+            std::vector< Dof<double>::Pointer > DoF;
+            DofsArrayType Doftemp = BasicTestSchemeDisplacement(model_part, pscheme, DoF, delta_time);
+            
+            CompressedMatrix A = ZeroMatrix(3, 3);
```

Comment:
here you are defining a FULL csr matrix. Is that what you want????

Reply:
It is just a test with a 3x3 matrix, I don't think we need to optimize that